src/detect-base64-data.c

```diff
@@ -67,47 +65,22 @@ static int g_file_data_buffer_id = 0;
 
 static int DetectBase64DataSetupTest01(void)
 {
-    DetectEngineCtx *de_ctx = NULL;
-    SigMatch *sm;
-    int retval = 0;
-
-    de_ctx = DetectEngineCtxInit();
-    if (de_ctx == NULL) {
-        goto end;
-    }
+    DetectEngineCtx *de_ctx = DetectEngineCtxInit();
+    FAIL_IF_NULL(de_ctx);
 
     de_ctx->flags |= DE_QUIET;
-    de_ctx->sig_list = SigInit(de_ctx,
+    Signature *s = DetectEngineAppendSig(de_ctx,
         "alert smtp any any -> any any (msg:\"DetectBase64DataSetupTest\"; "
         "base64_decode; base64_data; content:\"content\"; sid:1; rev:1;)");
-    if (de_ctx->sig_list == NULL) {
-        printf("SigInit failed: ");
-        goto end;
-    }
+    FAIL_IF_NULL(s);
 
-    sm = de_ctx->sig_list->init_data->smlists[DETECT_SM_LIST_PMATCH];
-    if (sm == NULL) {
-        printf("DETECT_SM_LIST_PMATCH should not be NULL: ");
-        goto end;
-    }
-    if (sm->type != DETECT_BASE64_DECODE) {
-        printf("sm->type should be DETECT_BASE64_DECODE: ");
-        goto end;
-    }
+    SigMatch *sm = de_ctx->sig_list->init_data->smlists[DETECT_SM_LIST_PMATCH];
```

Comment:
now that we have the `s` pointer, it's cleaner and clearer to use this here `SigMatch *sm = s->init_data->smlists[DETECT_SM_LIST_PMATCH];`